File: frontend/src/types/interview.ts
export const INTERVIEW_TYPES = [
  'Behavioral',
  'Technical',
  'System Design',
  'Recruiter Screen',
  'Final Round',
] as const

export const INTERVIEW_STATUSES = ['Scheduled', 'Completed', 'Cancelled'] as const

export interface MockQuestion {
  question: string
  tip: string
}

export type AIFeedbackStatus = 'ok' | 'not_configured' | 'error'

export interface InterviewSession {
  id: number
  application_id: number | null
  title: string
  interview_type: string
  scheduled_date: string | null
  status: string
  prep_notes: string | null
  performance_rating: number | null
  performance_notes: string | null
  mock_questions: MockQuestion[]
  ai_feedback_status: AIFeedbackStatus
  created_at: string
  updated_at: string
}

export interface InterviewSessionInput {
  title: string
  interview_type: string
  application_id: number | null
  scheduled_date: string | null
  prep_notes: string | null
}

export type InterviewSessionPatch = Partial<InterviewSessionInput> & {
  status?: string
  performance_rating?: number | null
  performance_notes?: string | null
}

File: backend/app/schemas/interview.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MockQuestionItem(BaseModel):
    question: str
    tip: str


class MockQuestionSet(BaseModel):
    questions: list[MockQuestionItem]


class InterviewSessionCreate(BaseModel):
    title: str
    interview_type: str = "Behavioral"
    application_id: int | None = None
    scheduled_date: str | None = None
    prep_notes: str | None = None


class InterviewSessionUpdate(BaseModel):
    title: str | None = None
    interview_type: str | None = None
    application_id: int | None = None
    scheduled_date: str | None = None
    status: str | None = None
    prep_notes: str | None = None
    performance_rating: int | None = None
    performance_notes: str | None = None


class InterviewSessionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    application_id: int | None
    title: str
    interview_type: str
    scheduled_date: str | None
    status: str
    prep_notes: str | None
    performance_rating: int | None
    performance_notes: str | None
    mock_questions: list[MockQuestionItem]
    ai_feedback_status: str
    created_at: datetime
    updated_at: datetime
